web/blog/tenzir-for-splunk-users/index.md

```diff
@@ -0,0 +1,316 @@
+  performs an explicit projection with the provided fields.
+
+### Bytes transferred over time by service
+
+Splunk:
+
+```splunk-spl
+index=zeek sourcetype="zeek_conn" OR sourcetype="zeek_conn_long"
+| eval orig_megabytes = round(orig_bytes/1024/1024,2)
+| eval resp_megabytes = round(resp_bytes/1024/1024,2)
+| eval orig_gigabytes = round(orig_megabytes/1024,2)
+| eval resp_gigabytes = round(resp_megabytes/1024,2)
+| timechart sum(orig_gigabytes) AS 'Outgoing Gigabytes',sum(resp_gigabytes) AS 'Incoming Gigabytes' by service span=1h
+```
+
+Tenzir:
+
+```splunk-spl
+export
+| where #schema == /zeek\.conn.*/
+| extend orig_megabytes=round(orig_bytes/1024/1024, 2)
+| extend resp_megabytes=round(orig_bytes/1024/1024, 2)
+| extend orig_gigabytes=round(orig_megabytes/1024, 2)
+| extend resp_gigabytes=round(orig_megabytes/1024, 2)
+| summarize 'Outgoing Gigabytes'=sum(orig_gigabytes), 'Incoming Gigabytes'=sum(resp_gigabytes) by service resolution 1h
+```
+
+This example also looks quite similar in structure.
+
+- This query spreads over two data sources: the event `zeek.conn` and
+  `zeek.conn_long`. The latter tracks long-running connections and is available
+  as [separate package](https://github.com/corelight/zeek-long-connections).
+
+- TQLs's `#schema` selector in an expression is responsible for filtering the
+  data sources. This is because all TQL pipelines are *multi-schema*, i.e., they
+  can process more than a single type of data.
+
+### Rare JA3 hashes
+
+Splunk:
+
+```splunk-spl
+index=zeek sourcetype=zeek_ssl
+| rare ja3
+```
+
+Tenzir:
+
+```splunk-spl
+export
+| where #schema == "zeek.ssl"
+| rare ja3
+| head 10
+```
+
+This example shows again how to select a specific data source and perform some
+"stack counting".
+
+### Expired certificates
+
+Splunk:
+
+```splunk-spl
+index=zeek sourcetype=zeek_x509
+| convert num(certificate.not_valid_after) AS cert_expire
+| eval current_time = now(), cert_expire_readable = strftime(cert_expire,"%Y-%m-%dT%H:%M:%S.%Q"), current_time_readable=strftime(current_time,"%Y-%m-%dT%H:%M:%S.%Q")
+| where current_time > cert_expire
+```
+
+Tenzir:
+
+```splunk-spl
+export
+| where certificate.not_valid_after > now()
+```
+
+This example shows the benefit of native time types (and Tenzir's rich type
+system in more generally).
+
+- TQL's [type system](/data-model/type-system) has first-class support for times
+  and durations.
+
+- TQL's [`zeek-tsv`](/formats/zeek-tsv) parser preserves `time` types natively,
+  so you don't have to massage strings at query time.
+
+### Large DNS queries
+
+Splunk:
+
+```splunk-spl
+index=zeek sourcetype=zeek_dns
+| eval query_length = len(query)
+| where query_length > 75
+| table _time id.orig_h id.resp_h proto query query_length answer
+```
+
+Tenzir:
+
+```splunk-spl
+export
+| extend query_length = length(zeek.dns.query)
+| where query_length > 75
+| select #timestamp, id.orig_h, id.resp_h, proto, query, query_length
```

Comment:
The `length(zeek.dns.query)` syntax is debatable. Perhaps you can use `where #schema == "zeek.dns"` as before?

Reply:
Fixed in be04a95fb2.